transforms/sort.go

```diff
@@ -0,0 +1,45 @@
+package transforms
+
+import (
+	"gopkg.in/Clever/optimus.v3"
+	"sort"
+)
+
+type rows struct {
+	less func(optimus.Row, optimus.Row) (bool, error)
+	rows []optimus.Row
+	err  error
+}
+
+func (r *rows) Len() int {
+	return len(r.rows)
+}
+func (r *rows) Less(i, j int) bool {
+	less, err := r.less(r.rows[i], r.rows[j])
+	if err != nil {
+		r.err = err
+	}
+	return less
+}
+func (r *rows) Swap(i, j int) {
+	r.rows[i], r.rows[j] = r.rows[j], r.rows[i]
+}
+
+// Sort takes in a function that reports whether the row i should sort before row j.
+// It outputs the rows in sorted order. The sort is not guaranteed to be stable.
+func Sort(less func(i, j optimus.Row) (bool, error)) optimus.TransformFunc {
```

Comment:
What do you think about offering stable sort as well?  You could just define a Stable transformer and then either add a flag to this function or have two functions (transforms.Sort and transforms.Stable) that both call out to the logic within the current Sort function.